packages/event-handler/src/rest/Route.ts

```diff
@@ -1,16 +1,28 @@
-import type { HttpMethod, Path, RouteHandler } from '../types/rest.js';
+import type {
+  HttpMethod,
+  Middleware,
+  Path,
+  RouteHandler,
+} from '../types/rest.js';
 
 class Route {
   readonly id: string;
   readonly method: string;
   readonly path: Path;
   readonly handler: RouteHandler;
+  readonly middleware: Middleware[];
 
-  constructor(method: HttpMethod, path: Path, handler: RouteHandler) {
+  constructor(
+    method: HttpMethod,
+    path: Path,
+    handler: RouteHandler,
+    middleware: Middleware[] = []
+  ) {
```

Comment:
Should we make this an object? It's starting to become unwieldy.

Reply:
Ah I see what you mean. I will do that in a separate PR but I agree, a new type would be nicer here.